tablecodec/tablecodec.go

```diff
@@ -312,11 +308,16 @@ func Unflatten(datum types.Datum, ft *types.FieldType) (types.Datum, error) {
 		return datum, nil
 	case mysql.TypeDate, mysql.TypeDatetime, mysql.TypeTimestamp:
 		var t mysql.Time
-		t.Type = ft.Tp
-		t.Fsp = ft.Decimal
-		err := t.Unmarshal(datum.GetBytes())
-		if err != nil {
-			return datum, errors.Trace(err)
+		var err error
+		if inIndex {
```

Comment:
We should add a new request type and check if  underlying kv store support this.